src/librbd/ImageWatcher.cc

```diff
@@ -1052,6 +1060,17 @@ void ImageWatcher<I>::handle_rewatch(int r) {
   CephContext *cct = m_image_ctx.cct;
   ldout(cct, 10) << this << " " << __func__ << ": r=" << r << dendl;
 
+  m_watch_error_count -= 1;
```

Comment:
The counter should be reset to zero in the ```rewatch``` method

Reply:
So, check for > 1, and always reset to 0, right? I wasn't sure whether multiple accumulated errors are fine to be squashed.

Reply:
Since a "rewatch" will effectively re-establish the watch again, it should be fine so long as you reset the counter before starting the rewatch. That way, if there is a failure during the re-watch, the counter will be incremented and it will attempt to rewatch again after the rewatch state machine has completed.